### PR TITLE
chore: add storage samples

### PR DESCRIPTION
**Pull request by TheRoyalTnetennba**

### CODE REVIEW

**Comment by quartzmo** [comment]
I'm guessing it's necessary to open this PR from a branch on this repo?

File: google-cloud-storage/Rakefile [hunk 1]
```diff
@@ -176,6 +176,40 @@ namespace :ci do
   end
 end
 
+task :samples do
+  Rake::Task["samples:latest"].invoke
+end
+
+namespace :samples do
+  task :latest do
+    if File.directory? "samples"
```
Comment (quartzmo) [inline google-cloud-storage/Rakefile:185]: Why check for the `samples` dir when we know it's there? (Same in `master` task.)

**Comment by quartzmo** [comment]
I pulled this branch and ran the samples locally, works for me.

File: google-cloud-storage/samples/acceptance/helper.rb [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  5.times do
+    begin
+      yield
+      return
+    rescue Google::Cloud::ResourceExhaustedError => e
+      puts "\n#{e} Gonna try again"
+      sleep rand(3..5)
+    rescue StandardError => e
+      puts "\n#{e}"
+      return
```
Comment (quartzmo) [inline google-cloud-storage/samples/acceptance/helper.rb:53]: Why rescue and puts `StandardError` here?

Comment (TheRoyalTnetennba) [reply google-cloud-storage/samples/acceptance/helper.rb]: debug thing I forgot to remove. removed.

File: google-cloud-storage/Rakefile (hunk 1 above)
Comment (TheRoyalTnetennba) [reply google-cloud-storage/Rakefile]: copied from the generated rakefile, removed

File: google-cloud-storage/samples/Gemfile [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  # rubocop:disable Bundler/DuplicatedGem
+  gem "google-cloud-kms"
+  # [START storage_dependencies]
+  gem "google-cloud-storage"
+  # [END storage_dependencies]
+  # rubocop:enable Bundler/DuplicatedGem
+end
+
+group :test do
+  gem "minitest", "~> 5.13"
```
Comment (dazuma) [inline google-cloud-storage/samples/Gemfile:33]: Is it possible to `~> 5.14`?

**Pull request merged**